Lesson11/Task2/Program.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+
+namespace Task2
+{
+    class Program
+    {
+        static private object block = new object();
+
+        static void Main(string[] args)
+        {
+            Thread[] threads = { new Thread(() => WriteTo("result.txt", ReadFrom("textA.txt"))),
+                                 new Thread(() => WriteTo("result.txt", ReadFrom("textB.txt")))
+                               };
+
+            for (int i = 0; i < threads.Length; i++)
+            {
+                threads[i].Start();
+            }
+
+            Console.ReadLine();
+        }
+
+        static void WriteTo(string filePath, string receivedText)
+        {
+
+            lock (block)
+            {
+
+                string resultLine = string.Format($"Writing result of thread {Thread.CurrentThread.ManagedThreadId} : {receivedText}");
```

Comment:
you should lock execution for the minimum needed amount of time so all operations that do not correspond to operating on the shared resources should  be moved out of the lock scope. like this string formatting